python/cuml/test/conftest.py

```diff
@@ -79,3 +84,54 @@ def pytest_collection_modifyitems(config, items):
         for item in items:
             if "stress" in item.keywords:
                 item.add_marker(skip_stress)
+
+
+@pytest.fixture(scope="module")
+def golden_option(request):
+    return request.config.getoption("--golden-values")
+
+@pytest.fixture(scope="module")
+def golden_module(request, golden_option):
+    gmodule = GoldenModule(request.node.name,
+                           os.path.dirname(request.node.fspath),
+                           golden_option)
+    yield gmodule
+    gmodule.cleanup()
+
+@pytest.fixture()
+def golden(request, golden_module):
+    """Allow golden (precomputed) values for PyTests
+
+    Golden values can be precomputed by one run, saved in JSON files, and
+    reused for comparisons in future runs.
+
+    Tests with long-running code should check on each run whether or not
+    to rerun the slow code by checking `golden.recompute`.
+
+    Attributes can be stored (when recomputing) on this golden object by
+    setting attributes that end with an underscore, like:
```

Comment:
I think using underscores in attribute names as some kind of interface is confusing. I would rather have something like `golden.save(name, value)` instead of `golden.name_ = value`.

Reply:
Fair point... we can always go more explicit here - same with the getter. It could be like `golden.set("accuracy", value)` and `golden.get("accuracy")`... I thought the implicit get/set would make code look a bit cleaner but I'm open to the change. Curious what others think.